### PR TITLE
Add Destination

### PR DESCRIPTION
**Pull request by voscob**
### Description

The Destination was added.

### Quick checks:

- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-cosmos-nosql/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: destination/destination.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			return i, fmt.Errorf("write record: %w", err)
+		}
+	}
+
+	return len(records), nil
+}
+
+// Teardown just logs the info event.
+func (d *Destination) Teardown(ctx context.Context) error {
+	sdk.Logger(ctx).Info().Msg("Tearing down an Azure Cosmos DB for NoSQL Destination")
```
Comment (maksenius) [inline destination/destination.go:85]: Could we close connection?

File: destination/writer.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+func (w *writer) create(ctx context.Context, record sdk.Record) error {
+	payload := make(sdk.StructuredData)
+	if err := json.Unmarshal(record.Payload.After.Bytes(), &payload); err != nil {
+		return fmt.Errorf("unmarshal payload: %w", err)
+	}
+
+	// if sdk.Record.Key does not contain the 'id' key
+	if _, okPayload := payload[common.KeyID]; !okPayload {
```
Comment (maksenius) [inline destination/writer.go:82]: we have comment `if sdk.Record.Key does not contain the 'id' key` but we check id in payload and not in key. Is it ok?

File: destination/writer.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+func (w *writer) replace(ctx context.Context, record sdk.Record) error {
+	payload := make(sdk.StructuredData)
+	if err := json.Unmarshal(record.Payload.After.Bytes(), &payload); err != nil {
+		return fmt.Errorf("unmarshal payload: %w", err)
+	}
+
+	// if sdk.Record.Key does not contain the 'id' key
+	if _, okPayload := payload[common.KeyID]; !okPayload {
```
Comment (maksenius) [inline destination/writer.go:113]: maybe lets move this code to separate function

File: destination/writer.go (hunk 2 above)
Comment (voscob) [reply destination/writer.go]: Yeah, you're right 👍 
Fixed!

File: destination/writer.go (hunk 3 above)
Comment (voscob) [reply destination/writer.go]: I've moved this logic to another method 👍

File: destination/destination.go (hunk 1 above)
Comment (voscob) [reply destination/destination.go:85]: Azure Cosmos DB package has no methods to close the connection :(

**Review by maksenius — APPROVED** [review]

**Pull request merged**